Library/Homebrew/formula.rb

```diff
@@ -956,33 +956,25 @@ def unlock
     @oldname_lock.unlock unless @oldname_lock.nil?
   end
 
+  def migration_needed?
+    oldname && !rack.exist? && (dir = HOMEBREW_CELLAR/oldname).directory? &&
+      !dir.subdirs.empty? && tap == Tab.for_keg(dir.subdirs.first).tap
```

Comment:
Could this be tweaked in a future PR so you `return false` for each of those cases rather than chaining with `&&`? It makes it much easier to follow.